server/src/main/java/org/opensearch/cluster/routing/WeightedRouting.java

```diff
@@ -25,6 +26,11 @@ public class WeightedRouting implements Writeable {
     private String attributeName;
     private Map<String, Double> weights;
 
+    public WeightedRouting() {
+        this.attributeName = "";
+        this.weights = new HashMap<>();
```

Comment:
Can we set the default HashMap to have 3 entries?

Reply:
Why do we need to create the instance without `attributeName `.

Reply:
This is to create empty weightedRouting during `delete` api call, since we are retaining version